MultiSource/Benchmarks/Ptrdist/anagram/anagram.c

```diff
@@ -499,15 +601,20 @@ jmp_buf jbAnagram;
         continue; \
     }
 
-
-void
-FindAnagram(Quad * pqMask, PPWord ppwStart, int iLetter)
+// CHECKEDC : function call formal parameters, checkedc array pointer
+// : pqMask declared bounds is from aqMainMask[MAX_QUADS]
+// : ppwStart declared bounds is from apwCand[MAXCAND]
+void FindAnagram(_Array_ptr<Quad> pqMask : count(MAX_QUADS),
+        PPWord ppwStart : bounds(apwCand, apwCand+MAXCAND), int iLetter)
 {
-    Quad aqNext[MAX_QUADS];
-    register PWord pw;
+    // CHECKEDC
+    Quad aqNext _Checked [MAX_QUADS];
+    register PWord pw = 0;
     Quad qMask;
     unsigned iq;
-    PPWord ppwEnd = &apwCand[0];
+    PPWord ppwEnd : bounds(apwCand, apwCand+MAXCAND) = &apwCand[0];
+    // CHECKEDC : pointer arithmetic, non-null check && in-bounds check
+    // dynamic_check((ppwEnd+cpwCand) >= apwCand && (ppwEnd+cpwCand) < (apwCand+MAXCAND))
```

Comment:
The meaning of pointer overflow in that section is similar to the meaning of integer overflow.  Creating a pointer larger than what can be represented in a pointer type is not allowed.     The check would use a processor's overflow detection (since at the processor level, this eventually involves integers addition).